### ISSUE TITLE
RAILS_FILTER_CLASS warning on Rails console

### ISSUE THREAD

**Issue by TonyArra** [newrelic/newrelic-ruby-agent#934]
Warnings for previous definition of `RAILS_FILTER_CLASS` is polluting the output for rails console.

## Description

Consistently receiving the following warnings when running a rails console locally with the newrelic_rpm gem.

```sh
rvm/gems/ruby-2.7.5@single-stop-api/gems/newrelic_rpm-8.4.0/lib/new_relic/agent/parameter_filtering.rb:14: 
warning: already initialized constant NewRelic::Agent::ParameterFiltering::RAILS_FILTER_CLASS

.rvm/gems/ruby-2.7.5@single-stop-api/gems/newrelic_rpm-8.4.0/lib/new_relic/agent/parameter_filtering.rb:14: 
warning: previous definition of RAILS_FILTER_CLASS was here
```

## Expected Behavior
No warnings shown in console output

## [Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results
Provide any other relevant log data.
TIP:  Scrub logs and diagnostic information for sensitive information 

## Steps to Reproduce

- Install newrelic_rpm:8.4.0 on a Rails 6.1 project.
- Run a `rails console`
- Observe warnings in output

## Your Environment
```sh
❯ ruby -v
ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [x86_64-darwin21]

❯ rails -v
Rails 6.1.4.4
```

## Additional context
Add any other context about the problem here. For example, relevant community posts or support tickets.

**Comment by kaylareopelle** [comment]
Thanks for bringing this to our attention, @TonyArra! We have not been able to replicate the exact error you describe, however, we intermittently saw a redefinition warning for another constant in NewRelic::Agent::ParameterFiltering when spinning up a Rails server on a Rails 6.1.4.4 app. 

We have a few ideas about how to resolve this issue and hope to get back to you with a branch to test the solution soon.

**Comment by tannalynn** [comment]
I've created a branch that should eliminate these warnings for the constants in the ParamterFiltering module. Since we are having a hard time reproducing the same warnings, you could try this branch out and then please let us know if you still see those same warnings, that would be helpful (@TonyArra and/or @rwojnarowski)

You can use this branch by updating the gem in your Gemfile to:
```
gem 'newrelic_rpm', :git => 'https://github.com/newrelic/newrelic-ruby-agent.git',  :branch => '934_constant_redefine_warning'
```

**Comment by TonyArra** [comment]
@tannalynn I can confirm for myself at least that I no longer see the warnings on that branch

**Comment by rwojnarowski** [comment]
@tannalynn can confirm. Fix working perfect, without warnings. Thanks!

**Comment by tannalynn** [comment]
Awesome! That's great to hear, thanks for letting me know! We'll get that merged and it will be included in our next release, 8.5.0.
